internal/nodes/avi_model_l4_translator.go

```diff
@@ -136,6 +137,11 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
 		buildWithInfraSetting(key, avi_vs_meta, vsVipNode, infraSetting)
 	}
 
+	// Copy the VS properties from L4Rule object
+	if l4Rule, err := getL4Rule(key, svcObj); err == nil {
+		buildWithL4Rule(key, avi_vs_meta, l4Rule)
```

Comment:
shared L4 VS need to take care.

Reply:
Shared VIP VS will be taken care of in the next PR as it requires testing and UT changes.
cc: @akshayhavile